File: packages/types/src/models/channel.ts
import { PaginatedResponse } from "../api";

/**
 * Channel model type definitions
 */

export interface Channel {
    channel_id: number;
    title: string | null;
    username: string | null;
    invite_link?: string | null;
    created_at: string;
    updated_at: string | null;
}

export interface ChannelResponse extends Channel {
    subscriber_count: number;
    linked_groups_count: number;
}

export interface ChannelGroupLink {
    group_id: number;
    title: string | null;
}

export interface ChannelDetailResponse extends ChannelResponse {
    linked_groups: ChannelGroupLink[];
}

export interface ChannelCreateRequest {
    channel_id: number;
    title?: string | null;
    username?: string | null;
    invite_link?: string | null;
}

export type ChannelListResponse = PaginatedResponse<ChannelResponse>;

File: apps/api/src/schemas/channel.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from .base import PaginatedResponse


class ChannelGroupLinkSchema(BaseModel):
    group_id: int
    title: str | None

    model_config = ConfigDict(from_attributes=True)


class ChannelBase(BaseModel):
    title: str | None
    username: str | None
    invite_link: str | None = None


class ChannelCreateRequest(ChannelBase):
    channel_id: int


class ChannelResponse(ChannelBase):
    channel_id: int
    created_at: datetime
    updated_at: datetime | None
    subscriber_count: int = 0  # Placeholder for live count
    linked_groups_count: int = 0

    model_config = ConfigDict(from_attributes=True)


class ChannelDetailResponse(ChannelResponse):
    linked_groups: list[ChannelGroupLinkSchema] = []


class ChannelListResponse(PaginatedResponse[ChannelResponse]):
    pass
